SparkleSpin/Controllers/PlayerViewController.swift

```diff
@@ -11,51 +11,97 @@ import UIKit
 class PlayerViewController: UIViewController {
     
     @IBOutlet weak var playerTableView: UITableView!
+    @IBOutlet weak var playerEntryTextField: CustomTextField!
+    @IBOutlet weak var addPlayerButton: LightButton! 
+    
+    let playerViewModel = PlayerViewModel()
+    var doneBarButton: UIBarButtonItem?
     
-
     override func viewDidLoad() {
         super.viewDidLoad()
         setNavigationBarTitle()
         registerNib()
         setupTableView()
-    
+        addDoneBarButtonItem()
     }
     
     private func setNavigationBarTitle() {
-        navigationItem.title = "Choose Players!"
+        navigationItem.title = "Add Players!"
+    }
+    
+    private func addDoneBarButtonItem() {
+        doneBarButton = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: nil)
+        navigationItem.rightBarButtonItem = doneBarButton
+        doneBarButton?.isEnabled = false
     }
     
     private func setupTableView() {
-        playerTableView.dataSource = self
+        playerTableView.dataSource = playerViewModel
         playerTableView.delegate = self
         playerTableView.backgroundColor = ThemeColor.Light.primaryColor
         playerTableView.separatorStyle = .none
         playerTableView.rowHeight = UITableView.automaticDimension
         playerTableView.estimatedRowHeight = CGFloat(Constants.estimatedRowHeight)
         playerTableView.tableFooterView = UIView()
     }
-
-}
-
-extension PlayerViewController: UITableViewDataSource, UITableViewDelegate {
     
-    func registerNib() {
+    private func registerNib() {
         let entryCell = UINib(nibName: NibID.entryCell, bundle: nil)
         playerTableView.register(entryCell, forCellReuseIdentifier: CellID.entryCell)
     }
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 1
+    private func addPlayerToList() {
+        let nameEntry = playerEntryTextField.text ?? ""
+        var playerToAdd = playerViewModel.createPlayerWith(name: nameEntry)
+        playerViewModel.addPlayerToSavedList(player: &playerToAdd)
+        playerEntryTextField.text? = ""
+    }
+    
+    private func updateDoneBarButton() {
+        if doneBarButton?.isEnabled == false {
+            doneBarButton?.isEnabled = true
+        }
```

Comment:
maybe this could be simplified with: `doneBarButton?.isEnabled = !doneBarButton?.isEnabled`?

Reply:
If this is changed in the method `updateDoneBarButton()`, and everything is as-is, there is a bug: On every other valid entry, the 'Done' button is disabled. I could add a check for the doneBarButton in `userTappedAddButton()`, but that will lead to a nested if-statement. 

Was there any other implementation you had in mind?